vfcompositor/Compositor.c

```
@@ -737,44 +772,104 @@ void OnMouseMove(int x, int y, int dx, int dy) {
 
         g_focused_window->rect.x = new_x;
         g_focused_window->rect.y = new_y;
+        g_focused_window->needs_redraw = true;
     }
 }
 
 void OnMouseButtonDown(int x, int y, uint8_t button) {
```

Clang (14.0.6)

[warning] function 'OnMouseButtonDown' has cognitive complexity of 54 (threshold 25)

(readability-function-cognitive-complexity)